File: Python/24. Swap Nodes in Pairs.py
# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next


class Solution:
    def swapPairs(self, head):
        dumm = ListNode()
        curr = dumm
        if head == None or head.next == None:
            return head

        while True:
            temp1 = ListNode(head.val)
            temp2 = ListNode(head.next.val, temp1)
            curr.next = temp2
            curr = curr.next.next
            head = head.next.next
            if head == None:
                return dumm.next

            elif head.next == None:
                curr.next = head
                return dumm.next

File: Javascript/24. Swap Nodes in Pairs.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var swapPairs = function(head) {
	let dumm = curr = new ListNode();
	if (head === null || head.next === null) return head;

	while (true) {
		const temp1 = new ListNode(head.val);
		const temp2 = new ListNode(head.next.val, temp1);

		curr.next = temp2;

		curr = curr.next.next;
		head = head.next.next;

		if (!head) {
			return dumm.next;
		} else if (!head.next) {
			curr.next = head;
			return dumm.next;
		}
	}
};